### PR TITLE
feat(design): new nav bar

### PR DESCRIPTION
**Pull request by vfried**
Implemented design from https://xd.adobe.com/view/161ea402-fc08-41c1-b52e-1c423af506c9-434d/

image

Incognito:
image

Collapsed
image

image

ToDo:
Clean up the css
Make a bottom line under the selected item if applies (Projects, Datasets, Environments)

/deploy

### CODE REVIEW

**Comment by lorenzo-cavazzi** [comment]
Nice to see the new design at work :rocket: 

I've noticed the footer may need to be adjusted. I can't scroll on Chrome (look at the bottom right in the screenshot) and it's not rendered at the bottom in Firefox (on the left).

Screenshot_20210310_185120

**Comment by vfried** [comment]
Yes, you are right about the footer.

I've also detected that but I will address the footer in a different PR so that it's easier to reviews.

**Review by lorenzo-cavazzi — CHANGES_REQUESTED** [review]
The content of the collapsable navbar doesn't look vertically centered (see screenshot).
I would reduce the top and the bottom margin on the renku logo so that the navbar height doesn't grow too much, but I'm also fine with the current solution.

I've seen you changed the `container` class. Should we create a different class instead if we need those CSS rules?
I would stick with the normal container since the current solution misaligns the content of the navbar and the main container on the left and on the right on a few pages.
I believe we haven't been consistent with the class elements for the layout (container --> col --> row) in many pages, and this is why the content is currently misaligned even with the default container (e.g. in the landing page we have 2 nested container).

Screenshot_20210311_085250

File: client/src/styles/bootstrap/_custom_bootstrap.scss [hunk 1]
```diff
@@ -2,11 +2,17 @@ $colors: map-merge($colors,$rk-colors);
...
 html {
-//	font-size: 14px;
-	// font-size: 0.90rem;
 	height:95%;
 }
 body { 
 	 height: 95%;
 }
 
+.container {
```
[the remark this answers is not in the document]
Comment (vfried) [reply client/src/styles/bootstrap/_custom_bootstrap.scss:11]: What you say on top is very true, i think we should change the way we are using containers. I'm taking a look at how we can use containers in a way that makes aligning things easier.

[the remark this answers is not in the document]
Comment (vfried) [reply client/src/styles/bootstrap/_custom_bootstrap.scss:11]: Also, yes... it's a bad idea to do that with container.

**Comment by vfried** [comment]
@ciyer @lorenzo-cavazzi I just pushed new changes, I was having lots of problems aligning elements in the bar so i decided to take a look at how bootstrap does the top navigation and did something similar to what they have. It seems to work. It's not 100% perfect but I think it's a big improvement.

We were having problems with the autosuggest and the way it was looking. Even in what we have deployed now there are problems. I also fixed this.

I also decided to delete all the containers and leave only two containers (top bar and body) let's see if this works, otherwise, we can add a few more. I hope we can slowly clean the pages from so many rows/cols/containers and things like that that make it hard to align items properly.

Let me know what you think about it.
It should get deployed soon.

**Comment by vfried** [comment]
image

image

image

---> I didn't add color to the top bar search because I had the feeling it was hard to read but I can add it and try a bit more to make it look better.

---> I could add a bit of padding to the right and left if we wanted it, but this gives us less space

---> bootstrap has something that looks nice with icons when the bar is collapsed, i could do it for us as well (see here https://getbootstrap.com/docs/5.0/components/navbar/) or on the image here...
image

--> the logged out bar will disappear so i didn't put too much effort on that (We could change the way the login button looks, maybe make it green if we wanted to)

**Comment by rokroskar** [comment]
This is looking great! Nice work, looking forward to having this live! 😄 🎉 
 
A few observations: 

* Looks like there are some issues with padding? 

image

image

* I don't know if that's in scope here but the color of the folder icon in the file browser needs to be updated:

image

* Is the dataset view narrower intentionally?

image

**Comment by vfried** [comment]
Hey @rokroskar thanks for your observations, these things are out of scope now but I will take them into account.

I am only addressing the top nav bar on this PR, we are splitting the work into smaller chunks so that it's easier to review.

**Comment by rokroskar** [comment]
Make sense, I was confused by the PR title since it seemed to implement everything already 😆

**Comment by pameladelgado** [comment]
general question about the new design: are we going to have a dark theme version of it available? from time to time some users ask if we have this option...

**Comment by vfried** [comment]
I changed the title to navbar i guess it makes more sense. 

We will be grouping all design related commits here https://github.com/SwissDataScienceCenter/renku-ui/pull/1274

@pameladelgado we will not have a dark theme at the moment but i think it's something that would be cool to have in the future.

**Comment by lorenzo-cavazzi** [comment]
@vfried thanks for looking into this.

The vertical alignment still seems a bit off, not sure if that is on purpose. This is what I get in Firefox.

Screenshot_20210315_155839

I played a bit with `margin: auto` and I was able to vertically center the components on the right.

Screenshot_20210315_155705

If this looks better, I can try to apply the changes to this PR

**Comment by vfried** [comment]
Hey, my intention was to throw the elements on the right to the bottom. I thought it looked better, but I'm not 100% sure about this. Maybe @ciyer can also pitch in and give his opinion.

**Comment by lorenzo-cavazzi** [comment]
Ah right, in the reference design the rightmost elements are vertically aligned with the logo
Screenshot_20210315_161520
While the search and the links are oddly aligned (I can't see the intention here since they are not aligned with any other element :thinking: )
Screenshot_20210315_161544

**Comment by ciyer** [comment]
I think there was a mistake made in the top nav alignment. Here is a comparison of top nav from the Renku UI XD file and the one from the earlier presentation version. The alignment in the presentation version looks better to my eyes.

image

I think we should do the alignment as it was in the presentation and check with the designer that that was the intention.

**Comment by vfried** [comment]
Things centered like said on top... ready to be reviewed again

See changes in the text input...
Option 1: padding-top 0rem --> i like this one
image

Option 2: padding-top:0.3rem
image

**Review by ciyer — APPROVED** [review]
Nice improvement. I like the way this is progressing!

**Pull request merged**